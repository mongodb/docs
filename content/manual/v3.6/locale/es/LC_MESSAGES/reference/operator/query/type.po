# SOME DESCRIPTIVE TITLE.
# Copyright (C) 2011-2014, MongoDB, Inc.
# This file is distributed under the same license as the mongodb-manual
# package.
#
# Translators:
msgid ""
msgstr ""
"Project-Id-Version: MongoDB Manual\n"
"Report-Msgid-Bugs-To: \n"
"POT-Creation-Date: 2019-03-19 11:30-0400\n"
"PO-Revision-Date: 2014-04-08 18:55+0000\n"
"Last-Translator: tychoish <tychoish@gmail.com>\n"
"Language: es\n"
"Language-Team: Spanish (http://www.transifex.com/projects/p/mongodb-"
"manual/language/es/)\n"
"Plural-Forms: nplurals=2; plural=(n != 1)\n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=utf-8\n"
"Content-Transfer-Encoding: 8bit\n"
"Generated-By: Babel 2.6.0\n"

#: ../source/reference/operator/query/type.txt:3
msgid "$type"
msgstr ""

#: ../source/reference/operator/query/type.txt:14
msgid "Definition"
msgstr ""

#: ../source/reference/operator/query/type.txt:18
msgid ""
":query:`$type` selects the documents where the *value* of the ``field`` "
"is an instance of the specified :term:`BSON` type(s). Querying by data "
"type is useful when dealing with highly unstructured data where data "
"types are not predictable."
msgstr ""

#: ../source/reference/operator/query/type.txt:23
msgid ""
"A :query:`$type` expression for a single :term:`BSON` type has the "
"following syntax:"
msgstr ""

#: ../source/reference/operator/query/type.txt:32
msgid ""
"You can specify either the number or alias for the :ref:`BSON type"
"<document-type-available-types>`"
msgstr ""

#: ../source/reference/operator/query/type.txt:35
msgid ""
"The :query:`$type` expression can also accept an array of :term:`BSON` "
"types and has the following syntax:"
msgstr ""

#: ../source/reference/operator/query/type.txt:42
msgid ""
"The above query will match documents where the ``field`` value is any of "
"the listed types. The types specified in the array can be either numeric "
"or string aliases."
msgstr ""

#: ../source/reference/operator/query/type.txt:46
msgid "See :ref:`document-querying-by-multiple-data-types` for an example."
msgstr ""

#: ../source/reference/operator/query/type.txt:48
msgid ""
":ref:`document-type-available-types` describes the BSON types and their "
"corresponding numeric and string aliases."
msgstr ""

#: ../source/reference/operator/query/type.txt:53
msgid ""
"If you wish to obtain the :ref:`BSON type <bson-types>` returned by an "
":ref:`operator expression <agg-quick-ref-operator-expressions>` rather "
"than filtering documents by their BSON type, use the :expression:`$type` "
"aggregation operator."
msgstr ""

#: ../source/reference/operator/query/type.txt:59
msgid "Behavior"
msgstr ""

#: ../source/reference/operator/query/type.txt:61
msgid ""
":query:`$type` returns documents where the BSON type of the ``field`` "
"matches the BSON type passed to :query:`$type`."
msgstr ""

#: ../source/reference/operator/query/type.txt:66
msgid ""
":query:`$type` now works with arrays in the same way it works with other "
"BSON types. Previous versions only matched documents where the ``field`` "
"contained a nested array."
msgstr ""

#: ../source/reference/operator/query/type.txt:73
msgid "Available Types"
msgstr ""

#: ../source/reference/operator/query/type.txt:75
msgid ""
":query:`$type` operator accepts string aliases for the BSON types in "
"addition to the numbers corresponding to the BSON types. Previous "
"versions only accepted the numbers corresponding to the BSON type."
msgstr ""

#: ../source/includes/fact-bson-types.rst:4
msgid "Type"
msgstr ""

#: ../source/includes/fact-bson-types.rst:5
msgid "Number"
msgstr ""

#: ../source/includes/fact-bson-types.rst:6
msgid "Alias"
msgstr ""

#: ../source/includes/fact-bson-types.rst:7
msgid "Notes"
msgstr ""

#: ../source/includes/fact-bson-types.rst:9
msgid "Double"
msgstr ""

#: ../source/includes/fact-bson-types.rst:10
msgid "1"
msgstr ""

#: ../source/includes/fact-bson-types.rst:11
msgid "\"double\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:14
msgid "String"
msgstr ""

#: ../source/includes/fact-bson-types.rst:15
msgid "2"
msgstr ""

#: ../source/includes/fact-bson-types.rst:16
msgid "\"string\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:19
msgid "Object"
msgstr ""

#: ../source/includes/fact-bson-types.rst:20
msgid "3"
msgstr ""

#: ../source/includes/fact-bson-types.rst:21
msgid "\"object\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:24
msgid "Array"
msgstr ""

#: ../source/includes/fact-bson-types.rst:25
msgid "4"
msgstr ""

#: ../source/includes/fact-bson-types.rst:26
msgid "\"array\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:29
msgid "Binary data"
msgstr ""

#: ../source/includes/fact-bson-types.rst:30
msgid "5"
msgstr ""

#: ../source/includes/fact-bson-types.rst:31
msgid "\"binData\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:34
msgid "Undefined"
msgstr ""

#: ../source/includes/fact-bson-types.rst:35
msgid "6"
msgstr ""

#: ../source/includes/fact-bson-types.rst:36
msgid "\"undefined\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:37
#: ../source/includes/fact-bson-types.rst:67
#: ../source/includes/fact-bson-types.rst:77
msgid "Deprecated."
msgstr ""

#: ../source/includes/fact-bson-types.rst:39
msgid "ObjectId"
msgstr ""

#: ../source/includes/fact-bson-types.rst:40
msgid "7"
msgstr ""

#: ../source/includes/fact-bson-types.rst:41
msgid "\"objectId\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:44
msgid "Boolean"
msgstr ""

#: ../source/includes/fact-bson-types.rst:45
msgid "8"
msgstr ""

#: ../source/includes/fact-bson-types.rst:46
msgid "\"bool\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:49
msgid "Date"
msgstr ""

#: ../source/includes/fact-bson-types.rst:50
msgid "9"
msgstr ""

#: ../source/includes/fact-bson-types.rst:51
msgid "\"date\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:54
msgid "Null"
msgstr ""

#: ../source/includes/fact-bson-types.rst:55
msgid "10"
msgstr ""

#: ../source/includes/fact-bson-types.rst:56
msgid "\"null\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:59
msgid "Regular Expression"
msgstr ""

#: ../source/includes/fact-bson-types.rst:60
msgid "11"
msgstr ""

#: ../source/includes/fact-bson-types.rst:61
msgid "\"regex\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:64
msgid "DBPointer"
msgstr ""

#: ../source/includes/fact-bson-types.rst:65
msgid "12"
msgstr ""

#: ../source/includes/fact-bson-types.rst:66
msgid "\"dbPointer\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:69
msgid "JavaScript"
msgstr ""

#: ../source/includes/fact-bson-types.rst:70
msgid "13"
msgstr ""

#: ../source/includes/fact-bson-types.rst:71
msgid "\"javascript\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:74
msgid "Symbol"
msgstr ""

#: ../source/includes/fact-bson-types.rst:75
msgid "14"
msgstr ""

#: ../source/includes/fact-bson-types.rst:76
msgid "\"symbol\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:79
msgid "JavaScript (with scope)"
msgstr ""

#: ../source/includes/fact-bson-types.rst:80
msgid "15"
msgstr ""

#: ../source/includes/fact-bson-types.rst:81
msgid "\"javascriptWithScope\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:84
#: ../source/reference/operator/query/type.txt:88
msgid "32-bit integer"
msgstr ""

#: ../source/includes/fact-bson-types.rst:85
msgid "16"
msgstr ""

#: ../source/includes/fact-bson-types.rst:86
msgid "\"int\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:89
msgid "Timestamp"
msgstr ""

#: ../source/includes/fact-bson-types.rst:90
msgid "17"
msgstr ""

#: ../source/includes/fact-bson-types.rst:91
msgid "\"timestamp\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:94
#: ../source/reference/operator/query/type.txt:90
msgid "64-bit integer"
msgstr ""

#: ../source/includes/fact-bson-types.rst:95
msgid "18"
msgstr ""

#: ../source/includes/fact-bson-types.rst:96
msgid "\"long\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:99
msgid "Decimal128"
msgstr ""

#: ../source/includes/fact-bson-types.rst:100
msgid "19"
msgstr ""

#: ../source/includes/fact-bson-types.rst:101
msgid "\"decimal\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:102
msgid "New in version 3.4."
msgstr ""

#: ../source/includes/fact-bson-types.rst:104
msgid "Min key"
msgstr ""

#: ../source/includes/fact-bson-types.rst:105
msgid "-1"
msgstr ""

#: ../source/includes/fact-bson-types.rst:106
msgid "\"minKey\""
msgstr ""

#: ../source/includes/fact-bson-types.rst:109
msgid "Max key"
msgstr ""

#: ../source/includes/fact-bson-types.rst:110
msgid "127"
msgstr ""

#: ../source/includes/fact-bson-types.rst:111
msgid "\"maxKey\""
msgstr ""

#: ../source/reference/operator/query/type.txt:83
msgid ""
":query:`$type` supports the ``number`` alias, which will match against "
"the following :term:`BSON` types:"
msgstr ""

#: ../source/reference/operator/query/type.txt:86
msgid "double"
msgstr ""

#: ../source/reference/operator/query/type.txt:92
msgid "decimal"
msgstr ""

#: ../source/reference/operator/query/type.txt:94
msgid "See :ref:`document-querying-by-data-type`"
msgstr ""

#: ../source/reference/operator/query/type.txt:97
msgid "MinKey and MaxKey"
msgstr ""

#: ../source/reference/operator/query/type.txt:99
msgid ""
":bsontype:`MinKey <data_minkey>` and :bsontype:`MaxKey <data_maxkey>` are"
" used in comparison operations and exist primarily for internal use. For "
"all possible :term:`BSON` element values, ``MinKey`` will always be the "
"smallest value while ``MaxKey`` will always be the greatest value."
msgstr ""

#: ../source/reference/operator/query/type.txt:104
msgid ""
"Querying for  ``minKey`` or ``maxKey`` with :query:`$type` will only "
"return fields that match the special ``MinKey`` or ``MaxKey`` values."
msgstr ""

#: ../source/reference/operator/query/type.txt:108
msgid ""
"Suppose that the ``data`` collection has two documents with ``MinKey`` "
"and ``MaxKey``:"
msgstr ""

#: ../source/reference/operator/query/type.txt:116
msgid "The following query will return the document with ``_id: 1``:"
msgstr ""

#: ../source/reference/operator/query/type.txt:122
msgid "The following query will return the document with ``_id: 2``:"
msgstr ""

#: ../source/reference/operator/query/type.txt:129
msgid "Examples"
msgstr ""

#: ../source/reference/operator/query/type.txt:134
msgid "Querying by Data Type"
msgstr ""

#: ../source/reference/operator/query/type.txt:136
msgid ""
"The ``addressBook`` contains addresses and zipcodes, where ``zipCode`` "
"has ``string``, ``int``, ``double``, and ``long`` values:"
msgstr ""

#: ../source/reference/operator/query/type.txt:151
msgid ""
"The following queries return all documents where ``zipCode`` is the "
":term:`BSON` type ``string``:"
msgstr ""

#: ../source/reference/operator/query/type.txt:160
#: ../source/reference/operator/query/type.txt:174
#: ../source/reference/operator/query/type.txt:187
msgid "These queries return:"
msgstr ""

#: ../source/reference/operator/query/type.txt:166
msgid ""
"The following queries return all documents where ``zipCode`` is the "
":term:`BSON` type ``double``:"
msgstr ""

#: ../source/reference/operator/query/type.txt:180
msgid ""
"The following query uses the ``number`` alias to return documents where "
"``zipCode`` is the :term:`BSON` type ``double``, ``int``, or ``long``:"
msgstr ""

#: ../source/reference/operator/query/type.txt:198
msgid "Querying by Multiple Data Type"
msgstr ""

#: ../source/reference/operator/query/type.txt:200
msgid ""
"The ``grades`` collection contains names and averages, where "
"``classAverage`` has ``string``, ``int``, and ``double`` values:"
msgstr ""

#: ../source/reference/operator/query/type.txt:214
msgid ""
"The following queries return all documents where ``classAverage`` is the "
":term:`BSON` type ``string`` or ``double``. The first query uses numeric "
"aliases while the second query uses string aliases."
msgstr ""

#: ../source/reference/operator/query/type.txt:223
msgid "These queries return the following documents:"
msgstr ""

#: ../source/reference/operator/query/type.txt:234
msgid "Querying by MinKey and MaxKey"
msgstr ""

#: ../source/reference/operator/query/type.txt:236
msgid ""
"The ``restaurants`` collection uses ``minKey`` for any grade that is a "
"failing grade:"
msgstr ""

#: ../source/reference/operator/query/type.txt:261
msgid "And ``maxKey`` for any grade that is the highest passing grade:"
msgstr ""

#: ../source/reference/operator/query/type.txt:285
msgid ""
"The following query returns any restaurant whose ``grades.grade`` field "
"contains ``minKey``:"
msgstr ""

#: ../source/reference/operator/query/type.txt:294
#: ../source/reference/operator/query/type.txt:328
msgid "This returns"
msgstr ""

#: ../source/reference/operator/query/type.txt:318
msgid ""
"The following query returns any restaurant whose ``grades.grade`` field "
"contains ``maxKey``:"
msgstr ""

#: ../source/reference/operator/query/type.txt:355
msgid "Querying by Array Type"
msgstr ""

#: ../source/reference/operator/query/type.txt:357
msgid "A collection named ``SensorReading`` contains the following documents:"
msgstr ""

#: ../source/reference/operator/query/type.txt:396
msgid ""
"The following query returns any document in which the ``readings`` field "
"is an array, empty or non-empty."
msgstr ""

#: ../source/reference/operator/query/type.txt:403
msgid "The above query returns the following documents:"
msgstr ""

#: ../source/reference/operator/query/type.txt:438
msgid ""
"In the documents with ``_id : 1``, ``_id : 2``, ``_id : 3``, and ``_id : "
"4``, the ``readings`` field is an array."
msgstr ""

#: ../source/reference/operator/query/type.txt:442
msgid "Additional Information"
msgstr ""

#: ../source/reference/operator/query/type.txt:444
msgid ""
":method:`~db.collection.find()`, :doc:`BSON Types </reference/bson-"
"types>`."
msgstr ""

# 23e9e551a2ad441ca9e1a50d260c74ec
#~ msgid "*Syntax*: ``{ field: { $type: <BSON type> } }``"
#~ msgstr ""

# fa3d55241c7048cdaaad0fc6609d1399
#~ msgid ""
#~ ":query:`$type` selects the documents where "
#~ "the *value* of the ``field`` is "
#~ "the specified :term:`BSON` type."
#~ msgstr ""

# 52569ba39ff84ad18b57ee178f5007b9
#~ msgid "Consider the following example:"
#~ msgstr ""

# 4e4c63aa9e7748578f331916f45a5446
#~ msgid ""
#~ "This query will select all documents "
#~ "in the ``inventory`` collection where "
#~ "the ``price`` field value is a "
#~ "Double."
#~ msgstr ""

# ff45173dff5946928563e5cce4f346bb
#~ msgid ""
#~ "If the ``field`` holds an array, "
#~ "the :query:`$type` operator performs the "
#~ "type check against the array elements"
#~ " and **not** the ``field``."
#~ msgstr ""

# 18f3fe2242c04653a3c197b9a4e8cfc9
#~ msgid "Consider the following example where the ``tags`` field holds an array:"
#~ msgstr ""

# 69692a695c5445968ecdd0f928a605bc
#~ msgid ""
#~ "This query will select all documents "
#~ "in the ``inventory`` collection where "
#~ "the ``tags`` array contains an element"
#~ " that is itself an array."
#~ msgstr ""

# f4f40b49d0e14394aa5c00f3cf5779ac
#~ msgid ""
#~ "If instead you want to determine "
#~ "whether the ``tags`` field is an "
#~ "array type, use the :query:`$where` "
#~ "operator:"
#~ msgstr ""

# d4bb9a3a13814faea51c81874d9c599d
#~ msgid "See the :issue:`SERVER-1475` for more information about the array type."
#~ msgstr ""

# c062fd7f91c345f8a86711a5cd243b67
#~ msgid ""
#~ "Refer to the following table for "
#~ "the available :term:`BSON` types and "
#~ "their corresponding numbers."
#~ msgstr ""

# 6ba1a306440a4f7e9d2d2b81e121da6d
#~ msgid "**Type**"
#~ msgstr ""

# b3ae2e06af2541939888815932afa5e6
#~ msgid "**Number**"
#~ msgstr ""

# 897f67d9539646118c5dfafa5792684f
#~ msgid "Undefined (deprecated)"
#~ msgstr ""

# ce6fcf68778442aa9f9d2a421a8d6e3e
#~ msgid "Object id"
#~ msgstr ""

# 676c2639fd2c4ae69348b0efa9cd9ab6
#~ msgid "255"
#~ msgstr ""

# 08c41e855d9448108a6d320e819489e1
#~ msgid ""
#~ "``MinKey`` and ``MaxKey`` compare less "
#~ "than and greater than all other "
#~ "possible :term:`BSON` element values, "
#~ "respectively, and exist primarily for "
#~ "internal use."
#~ msgstr ""

# de71905419aa4af99b798d143f24ef2b
#~ msgid ""
#~ "To query if a field value is "
#~ "a ``MinKey``, you must use the "
#~ ":query:`$type` with ``-1`` as in the "
#~ "following example:"
#~ msgstr ""

# 6083273e1aec4e8096988088dbc552a6
#~ msgid "Example"
#~ msgstr ""

# 52562d3d349d49e88be5b501287ab5ea
#~ msgid ""
#~ "Consider the following example operation "
#~ "sequence that demonstrates both type "
#~ "comparison *and* the special ``MinKey`` "
#~ "and ``MaxKey`` values:"
#~ msgstr ""

# 63b729f5015e45a7a07a889e334bc950
#~ msgid ""
#~ "To query for the minimum value of"
#~ " a :term:`shard key` of a "
#~ ":term:`sharded cluster`, use the following "
#~ "operation when connected to the "
#~ ":program:`mongos`:"
#~ msgstr ""

# 6811d8ab85d64507943ac590ef13e00d
#~ msgid ""
#~ "Storing values of the different types"
#~ " in the same field in a "
#~ "collection is *strongly* discouraged."
#~ msgstr ""

# 2c1e59c3232e4d0585ebb4b075413994
#~ msgid ""
#~ ":method:`~db.collection.find()`, "
#~ ":method:`~db.collection.insert()`, :query:`$where`, "
#~ ":term:`BSON`, :term:`shard key`, :term:`sharded "
#~ "cluster` ."
#~ msgstr ""

