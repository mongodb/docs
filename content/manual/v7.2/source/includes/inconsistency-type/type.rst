
Type of inconsistency document.  Returns a value of |incon-type| for this   
inconsistency type.


