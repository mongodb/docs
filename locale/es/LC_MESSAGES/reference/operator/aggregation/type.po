# SOME DESCRIPTIVE TITLE.
# Copyright (C) 2008-2016
# This file is distributed under the same license as the mongodb-manual
# package.
# FIRST AUTHOR <EMAIL@ADDRESS>, 2019.
#
#, fuzzy
msgid ""
msgstr ""
"Project-Id-Version: mongodb-manual 3.4\n"
"Report-Msgid-Bugs-To: \n"
"POT-Creation-Date: 2016-12-08 12:02-0500\n"
"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
"Language-Team: LANGUAGE <LL@li.org>\n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=utf-8\n"
"Content-Transfer-Encoding: 8bit\n"
"Generated-By: Babel 2.6.0\n"

# d4bbf988f7974f2483b5728ffd72c3bf
#: ../source/reference/operator/aggregation/type.txt:3
msgid "$type (aggregation)"
msgstr ""

# b64bd8c1692c4eea99f174e663e70172
#: ../source/reference/operator/aggregation/type.txt
msgid "On this page"
msgstr ""

# e192009c6d8d4f9eb3dd791386a8525e
#: ../source/reference/operator/aggregation/type.txt:14
msgid "Definition"
msgstr ""

# 7d402f99fec8447ba373533442270be5
#: ../source/reference/operator/aggregation/type.txt:20
msgid ""
"Returns a string that specifies the :doc:`BSON type </reference/bson-"
"types/>` of the argument."
msgstr ""

# f945fd8c35f04e5aae6685fce1744a62
#: ../source/reference/operator/aggregation/type.txt:23
msgid ""
":expression:`$type` has the following :ref:`operator expression syntax "
"<agg-quick-ref-operator-expressions>`:"
msgstr ""

# 588f2206f51b43a5836b5161aa92cc0e
#: ../source/reference/operator/aggregation/type.txt:30
msgid "The argument can be any valid :ref:`expression <aggregation-expressions>`."
msgstr ""

# cc6ad411c27d468d97006b1a03f6cd20
#: ../source/reference/operator/aggregation/type.txt:34
msgid "Behavior"
msgstr ""

# b0cfa8014aac45ba812f016c4f0ae09c
#: ../source/reference/operator/aggregation/type.txt:36
msgid ""
"Unlike the :query:`$type` query operator, which matches array elements "
"based on their BSON type, the :expression:`$type` aggregation operator "
"does not examine array elements. Instead, when passed an array as its "
"argument, the ``$type`` aggregation operator returns the type of the "
"argument, i.e. ``\"array\"``."
msgstr ""

# 22f64d3123a3458a9400f3d2226e30b2
#: ../source/reference/operator/aggregation/type.txt:42
msgid ""
"If the argument is a field that is missing in the input document, "
":expression:`$type` returns the string ``\"missing\"``."
msgstr ""

# 3e4e567ed44a4eb99009492a9464a867
# bd517e8c1ac048f68984aa7b059b2a5e
#: ../source/reference/operator/aggregation/type.txt:53
#: ../source/reference/operator/aggregation/type.txt:89
msgid "Example"
msgstr ""

# 6bc0fb9c1be7482dbb2742b74c550a4d
#: ../source/reference/operator/aggregation/type.txt:54
msgid "Results"
msgstr ""

# ce53f805dce046ba9fadfbc807a17762
#: ../source/reference/operator/aggregation/type.txt:56
msgid "``{ $type: \"a\" }``"
msgstr ""

# eae4710260ac4f8e9ff0f2229752766c
#: ../source/reference/operator/aggregation/type.txt:57
msgid "``\"string\"``"
msgstr ""

# 66f62df1c9a14c01848db22f2c103136
#: ../source/reference/operator/aggregation/type.txt:59
msgid "``{ $type: /a/ }``"
msgstr ""

# 780556e0e1554dedbe1327da6ba07595
#: ../source/reference/operator/aggregation/type.txt:60
msgid "``\"regex\"``"
msgstr ""

# 4c60f0da177842f18367b5c7559507a6
#: ../source/reference/operator/aggregation/type.txt:62
msgid "``{ $type: 1 }``"
msgstr ""

# 1d3a927d71a94a06b4915840248d7a47
#: ../source/reference/operator/aggregation/type.txt:63
msgid "``\"double\"``"
msgstr ""

# fbd0acd7b10d45f4bcadf4b541747e2a
#: ../source/reference/operator/aggregation/type.txt:65
msgid "``{ $type: NumberLong(627) }``"
msgstr ""

# 0a5784fcb9e84d30b51cccab482a9a56
#: ../source/reference/operator/aggregation/type.txt:66
msgid "``\"long\"``"
msgstr ""

# 1f9af95d830c46c98a56437aafde2fe4
#: ../source/reference/operator/aggregation/type.txt:68
msgid "``{ $type: { x: 1 } }``"
msgstr ""

# d31c5584126f4fca8051223ac97ba04a
#: ../source/reference/operator/aggregation/type.txt:69
msgid "``\"object\"``"
msgstr ""

# 7246de5f63514059b431ce73400c951c
#: ../source/reference/operator/aggregation/type.txt:71
msgid "``{ $type: [ [ 1, 2, 3 ] ] }``"
msgstr ""

# b10cb37f321a45aca88ef60ed5c296ef
#: ../source/reference/operator/aggregation/type.txt:72
msgid "``\"array\"``"
msgstr ""

# eb2195ed901d4c3897c761dbca618177
#: ../source/reference/operator/aggregation/type.txt:76
msgid ""
"In the case of a literal array such as ``[ 1, 2, 3 ]``, enclose the "
"expression in an outer set of array brackets to prevent MongoDB from "
"parsing ``[ 1, 2, 3 ]`` as an :ref:`argument list <agg-quick-ref-"
"operator-expressions>` with three arguments (``1, 2, 3``). Wrapping the "
"array ``[ 1, 2, 3 ]`` in a :expression:`$literal` expression achieves the"
" same result."
msgstr ""

# 2eb5e505de2240d388cc7fdb1ac7adea
#: ../source/reference/operator/aggregation/type.txt:85
msgid ""
"See :ref:`operator expression syntax forms <agg-quick-ref-operator-"
"expressions>` for more information."
msgstr ""

# 4234818fa20a46328f7e67ebc05f2177
#: ../source/reference/operator/aggregation/type.txt:91
msgid ""
"This example uses a :term:`collection <collection>` named ``coll`` with "
"the following :term:`documents <document>`:"
msgstr ""

# c0eeac8b6795478988661a9c885d1db7
#: ../source/reference/operator/aggregation/type.txt:103
msgid ""
"The following aggregation operation uses the ``$type`` operator to "
"display the type of field ``a`` for all documents as part of the "
":pipeline:`$project` stage."
msgstr ""

# e5ac12ef5757452eb4e348cde61611df
#: ../source/reference/operator/aggregation/type.txt:115
msgid "The operation returns the following:"
msgstr ""

