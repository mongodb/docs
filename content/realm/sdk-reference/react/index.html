<!doctype html>
<head>
  <meta http-equiv="refresh" content="0; url=https://mongodb.com/docs/realm-sdks/react/latest" />
</head>
<body>
  <p><a href="https://mongodb.com/docs/realm-sdks/react/latest">Redirect</a></p>
</body>
