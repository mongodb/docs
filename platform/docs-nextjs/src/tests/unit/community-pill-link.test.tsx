import { render } from '@testing-library/react';
import CommunityPillLink from '@/components/community-pill-link';

// data for this component
import mockData from '../data/community-pill-link.test.json';
import type { CommunityDriverPill } from '@/types/ast';

it('community drivers directives render correctly', () => {
  const { argument, options } = mockData as CommunityDriverPill;
  const tree = render(<CommunityPillLink argument={argument} options={options} />);
  expect(tree.asFragment()).toMatchSnapshot();
});
