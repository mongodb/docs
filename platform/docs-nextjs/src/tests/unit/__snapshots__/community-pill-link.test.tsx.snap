// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`community drivers directives render correctly 1`] = `
<DocumentFragment>
  .emotion-0:last-of-type {
  margin-bottom: 16px;
}

.emotion-1 {
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  display: inline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-decoration-color: transparent;
  cursor: pointer;
  font-size: inherit;
  line-height: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: none;
  border: none;
  padding: 0;
  font-size: 13px;
  line-height: 20px;
  color: #016BF8;
  font-weight: 400;
  display: inline;
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

.emotion-1:hover,
.emotion-1[data-hover='true'],
.emotion-1:focus-visible,
.emotion-1[data-focus='true'] {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-transition: text-decoration 150ms ease-in-out;
  transition: text-decoration 150ms ease-in-out;
  text-underline-offset: 4px;
  text-decoration-thickness: 2px;
}

.emotion-1:focus {
  outline: none;
}

.emotion-1:hover,
.emotion-1[data-hover='true'] {
  text-decoration-color: #E8EDEB;
}

.emotion-1:focus-visible,
.emotion-1[data-focus='true'] {
  text-decoration-color: #016BF8;
}

.emotion-1>span>code,
.emotion-1>code {
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

.emotion-2 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: relative;
  bottom: 2px;
  left: -1px;
  height: 12px;
}

.emotion-3 {
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 700;
  font-size: 12px;
  line-height: 16px;
  border-radius: 24px;
  height: 18px;
  padding-left: 6px;
  padding-right: 6px;
  text-transform: uppercase;
  border: 1px solid;
  letter-spacing: 0.4px;
  background-color: #F9FBFA;
  border-color: #E8EDEB;
  color: #5C6C75;
}

<div
    class="emotion-0"
  >
    <a
      class="lg-ui-0001 emotion-1"
      href="https://example.com"
      rel="noopener noreferrer"
    >
      <span>
        MongoEngine for Flask
      </span>
      <svg
        alt=""
        aria-hidden="true"
        class="emotion-2"
        height="16"
        role="presentation"
        viewBox="0 0 16 16"
        width="16"
      >
        <path
          d="M13.823 2.4491C13.8201 2.30008 13.6999 2.17994 13.5509 2.17704L9.5062 2.09836C9.25654 2.09351 9.12821 2.39519 9.30482 2.5718L10.3856 3.65257L7.93433 6.10383C7.87964 6.15852 7.83047 6.21665 7.78683 6.27752L5.99909 8.06525C5.46457 8.59977 5.46457 9.4664 5.99909 10.0009C6.53361 10.5354 7.40023 10.5354 7.93475 10.0009L9.72249 8.21317C9.78336 8.16953 9.84148 8.12037 9.89618 8.06567L12.3474 5.61441L13.4282 6.69518C13.6048 6.87179 13.9065 6.74347 13.9016 6.4938L13.823 2.4491Z"
          fill="currentColor"
        />
        <path
          d="M7.25 3.12893C7.66421 3.12893 8 3.46472 8 3.87893C8 4.29315 7.66421 4.62893 7.25 4.62893H4C3.72386 4.62893 3.5 4.85279 3.5 5.12893V11.9929C3.5 12.2691 3.72386 12.4929 4 12.4929H10.864C11.1401 12.4929 11.364 12.2691 11.364 11.9929V8.75C11.364 8.33579 11.6998 8 12.114 8C12.5282 8 12.864 8.33579 12.864 8.75V11.9929C12.864 13.0975 11.9686 13.9929 10.864 13.9929H4C2.89543 13.9929 2 13.0975 2 11.9929V5.12893C2 4.02436 2.89543 3.12893 4 3.12893H7.25Z"
          fill="currentColor"
        />
      </svg>
    </a>
    <div
      class="emotion-3"
    >
      community built
    </div>
  </div>
</DocumentFragment>
`;
